index.js

```diff
@@ -1,26 +1,68 @@
 'use strict';
 const path = require('path');
-const pathExists = require('path-exists');
+const fs = require('fs');
+const {promisify} = require('util');
 const pLocate = require('p-locate');
 
+const fsStat = promisify(fs.stat);
+const fsLStat = promisify(fs.lstat);
+
+const typeMappings = {
+	blockDevice: 'isBlockDevice',
+	characterDevice: 'isCharacterDevice',
+	directory: 'isDirectory',
+	fifo: 'isFIFO',
+	file: 'isFile',
+	socket: 'isSocket',
+	symbolicLink: 'isSymbolicLink'
```

Comment:
I can't really imagine anyone needing to use `locate-path` to find anything other than `file` or `directory`. This seems like overkill.